lib/user_agent/browsers/pocket_casts.rb

```diff
@@ -0,0 +1,98 @@
+class UserAgent
+  module Browsers
+    # Pocket Casts BMID/E678F58F21
+    # PocketCasts/1.0 (Pocket Casts Feed Parser; +http://pocketcasts.com/)
+    # Shifty Jelly Pocket Casts, Android v4.5.3
+    # Shifty Jelly Pocket Casts, iOS v4.3
+    # Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Pocket Casts/1.1 Pocket Casts/1.1
+    class PocketCasts < Base
+      ANDROID_REGEX            = /Android/
+      IOS_REGEX                = /iOS/
+      POCKETCASTS_REGEX        = /PocketCasts/
+      POCKET_CASTS_REGEX       = /Pocket Casts/
+      POCKET_CASTS_SLASH_REGEX = /Pocket Casts\//
+      WINDOWS_REGEX            = /Windows/
+      WINDOWS_NT_REGEX         = /Windows NT/
+
+      def self.extend?(agent)
+        agent.detect { |useragent| POCKETCASTS_REGEX.match?(useragent.product) } || POCKET_CASTS_REGEX.match?(agent.to_s)
+      end
+
+      def browser
+        'Pocket Casts'
+      end
+
+      # Gets the right application
+      def application
+        detect_product('PocketCasts')
+      end
+
+      # Gets the operating system
+      #
+      # @return [String] the os
+      def os
+        application = reject { |agent| agent.comment.nil? || agent.comment.empty? }.first
```

Comment:
use a different variable name here since there is a method `application` already which you are hiding